src/foam/core/Action.js

```diff
@@ -134,59 +134,58 @@ foam.CLASS({
   ],
 
   methods: [
-    function andSlots(a, b) {
-      return this.ExpressionSlot.create({
-        args: [ a, b ],
-        code: function(a, b) {
-          return a && b;
-        }
-      });
-    },
 
-    function andSlotAndPromise(slot, promise) {
-      return this.andSlots(slot, this.PromiseSlot.create({
-        promise: promise
-      }));
+    function hasPermissions_(x, property) {
```

Comment:
Naming the second argument 'property' is a bit confusing because I would then expect it to be a Property. Maybe 'fn' would be a better name.